grafana/grafana
# Prometheus: Add hints to query builder

**What this PR does / why we need it**:
This PR adds hinting to Query builder. We are reusing hinting used in Query field. We will have a **UX ideation session** for improving the UX, but in the meantime I thought that we could add it and implement UI changes in separate PR. 

https://user-images.githubusercontent.com/30407135/153586190-e8aec4fd-fc30-4adc-a146-7ccf994f6469.mov

I have also changed a copy a little so it is more reusable. @achatterjee-grafana could you please have a look. 👀

**Which issue(s) this PR fixes**:
Part of https://github.com/grafana/grafana/issues/44253

**To test:**

1. Enable `promQueryBuilder` feature toggle in `custom.ini`
2. In metric dropdown choose a metric:
  - that includes `_bucket` in name and `hint: add histogram_qunatile()` should be visible. When you click, histogram quantile aggregation operations are added.
  -  that includes `_sum` or `_count` and  `hint: add rate() should be visible.  When you click, histogram quantile aggregation operations are added.
  - that returns a lot of series (21+). Run query with selected metric. `hint: add sum()` should be visible.
